### PR TITLE
BVAL-496 Spec changes around @Past/@Future support for JSR 310 date/time types

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: sources/constraint-definition.asciidoc [hunk 1]
```diff
@@ -699,544 +699,13 @@ The [classname]`ConstraintValidatorContext` object passed to the [methodname]`is
...
-             */
-            ConstraintValidatorContext addConstraintViolation();
-        }
-    }
-}
+include::{validation-api-source-dir}javax/validation/ConstraintValidatorContext.java[lines=11..-1]
 ----
 
-The [classname]`ConstraintValidatorContext` interface allows redefinition of the default constraint message generated when a constraint is not valid. [tck-testable]#By default, each invalid constraint leads to the generation of one error object represented by a [classname]`ConstraintViolation` object. This object is built from the default constraint message template as defined by the constraint declaration and the context in which the constraint declaration is placed (bean, property, executable parameter, cross-parameter or executable return value).#
+The [classname]`ConstraintValidatorContext` interface provides access to contextual information useful for the validation of specific constraints (`getClockProvider()`, see <<constraintsdefinitionimplementation-validationimplementation-temporal-validators>> ).
```
Comment (gsmet) [inline sources/constraint-definition.asciidoc]: Maybe "for instance `getClockProvider()`"?

File: sources/constraint-definition.asciidoc [hunk 2]
```diff
@@ -1331,6 +800,25 @@ context.buildConstraintViolationWithTemplate(
...
+
+[tck-testable]#
+Constraint validators for temporal constraints (either the built-in constraints `@Past` and `@Future` or custom temporal constraints) can obtain the current instant from the `ClockProvider` object exposed by `ConstraintValidatorContext#getClockProvider()`.#
+
+[source, JAVA]
+----
+include::{validation-api-source-dir}javax/validation/ClockProvider.java[lines=11..-1]
+----
+
+[tck-testable]#The `getClock()` method returns a `java.time.Clock` object which represents the current instant, date and time using a time-zone.
```
Comment (gsmet) [inline sources/constraint-definition.asciidoc]: s/time-zone/time zone/

File: sources/validation-api.asciidoc [hunk 3]
```diff
@@ -2119,7 +2119,7 @@ factory.close();
 
 The second example shows how a container can customize some Bean Validator resource handling to match its own behavior.
 
-.Customize message resolution, traversable resolver, constraint Validator factory and parameter name provider implementation
+.Customize message resolution, traversable resolver, constraint Validator factory parameter name provider and clock provider implementation
```
Comment (gsmet) [inline sources/validation-api.asciidoc]: missing comma after factory

File: sources/validation-api.asciidoc [hunk 4]
```diff
@@ -2374,90 +2280,21 @@ acmeFactory.setSpecificConfiguration( [...] );
...
 
-[classname]`ValidatorContext` returned by [methodname]`usingContext()` can be used to customize the state in which the [classname]`Validator` must be initialized. This is used to customize the [classname]`MessageInterpolator`, the [classname]`TraversableResolver`, [classname]`ParameterNameProvider` or the [classname]`ConstraintValidatorFactory`.
+[tck-testable]#[methodname]`getClockProvider()` returns the [classname]`ClockProvider` instance configured during the initialization of the [classname]`ValidatorFactory`.# It is particularly useful to build a [classname]`Validator` specific [classname]`ClockProvider` wrapping the one from the [classname]`ValidatorFactory`.
+
+[classname]`ValidatorContext` returned by [methodname]`usingContext()` can be used to customize the state in which the [classname]`Validator` must be initialized. This is used to customize the [classname]`MessageInterpolator`, the [classname]`TraversableResolver`, [classname]`ParameterNameProvider`, [classname]`ClockProvider` or the [classname]`ConstraintValidatorFactory`.
```
Comment (gsmet) [inline sources/validation-api.asciidoc]: 2 missing "the".

Btw, the new guidelines are to get rid of `[classname]` and `methodname`. At least for the doc of OGM, Search and HV. I suppose it applies to the spec too.

File: sources/constraint-definition.asciidoc [hunk 5]
```diff
@@ -1331,6 +800,25 @@ context.buildConstraintViolationWithTemplate(
...
+A conforming Bean validation implementation provides a default clock provider which returns a clock representing the current system time and default time zone.#
```
Comment (gsmet) [inline sources/constraint-definition.asciidoc]: Bean **V**alidation

File: sources/validation-api.asciidoc (hunk 4 above)
Comment (gunnarmorling) [inline sources/validation-api.asciidoc]: Ah yes. I generally don't add it, here it was more c&p and staying consistent with the other usages in the same sentence.

**Comment by gunnarmorling** [comment]
Pushed another commit for addressing @gsmet's review remarks.

**Comment by gsmet** [comment]
Amended the last commit to include the issue number and merged. Thanks!

**Comment by gunnarmorling** [comment]
Closing, as we heard no further feedback.

**Pull request closed without merging**